src/python/grpcio_tests/tests/testing/_application_testing_common.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2017 The gRPC Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import grpc_testing
+
+from tests.testing.proto import requests_pb2
+from tests.testing.proto import services_pb2
+
+_FIRST_SERVICE_UNUN = grpc_testing.method('UnUn', requests_pb2.Up,
```

Comment:
I agree too, but [Protocol Buffers Python is currently missing support for service descriptors adequately describing services](https://github.com/google/protobuf/issues/3168). (This is why the double creation function is named `channel_fixture_from_descriptions` rather than just `channel_fixture`; it leaves room for the much better `channel_fixture_from_descriptors` in the future.)

To dig into the example: the application and its author don't "have a protobuf stub"; they have a protobuf stub _class_, and instantiation of that class requires a `grpc.Channel`, and if they "want to mock server responses" at the _message_ level, not the level of serialized messages, that means that the test infrastructure has to be told at `grpc.Channel`-instantiation-time about which methods have which cardinality and which request and response classes. Until the feature is complete in Protocol Buffers Python and "the methods _[can]_ come from the service protos", I think this is the best way.